### PR TITLE
Migration: HyperV: 6 vhdx resize TCs

### PR DESCRIPTION
**Pull request by santoshx**

### CODE REVIEW

**Comment by santoshx** [comment]
$ cat TestResults/2018-11-10-19-16-51-6878/Logs.txt
10/11/2018 13:46:51 : [INFO ] Created LogDir: .\TestResults\2018-11-10-19-16-51-687810/11/2018 13:46:51 : [INFO ] 7za.exe file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] dos2unix.exe file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] gawk file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] jq file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] plink.exe file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] pscp.exe file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] kvp_client32 file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] kvp_client64 file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] nc.exe file found in Tools folder.
10/11/2018 13:46:51 : [INFO ] Test parameters have been validated successfully. Continue running the test.
10/11/2018 13:46:51 : [INFO ] Read Test Location from GlobalConfiguration.Global.HyperV.Hosts.ChildNodes[0].ServerName
10/11/2018 13:46:51 : [INFO ] Validating XML Files from D:\Santosh\GitHub\LISAv2 folder recursively...
10/11/2018 13:46:52 : [INFO ] Collected VHDX-RESIZE-TO-BIGGER512
10/11/2018 13:46:52 : [INFO ] Auto created D:\Santosh\GitHub\LISAv2\TestConfiguration.xml validated successfully.
10/11/2018 13:46:52 : [INFO ] .\AutomationManager.ps1 -xmlConfigFile 'D:\Santosh\GitHub\LISAv2\TestConfiguration.xml' -cycleName 'TC-33821' -RGIdentifier 'santoshx' -runtests -UseAzureResourceManager
10/11/2018 13:46:52 : [INFO ] '.\TestResults\2018-11-10-19-16-51-6878' saved to .\report\lastLogDirectory.txt
10/11/2018 13:46:52 : [INFO ] Info : AzureAutomationManager.ps1 - LIS on Azure Automation
10/11/2018 13:46:52 : [INFO ] Info : Created test results directory:.\TestResults\2018-11-10-19-16-51-6878
10/11/2018 13:46:52 : [INFO ] Info : Using config file D:\Santosh\GitHub\LISAv2\TestConfiguration.xml
10/11/2018 13:46:52 : [INFO ] HyperV Host            : localhost
10/11/2018 13:46:52 : [INFO ] Source VHD Path        : D:\VHDs\
10/11/2018 13:46:52 : [INFO ] Destination VHD Path   : VHDs_Destination_Path
10/11/2018 13:46:52 : [INFO ] Starting the Cycle - TC-33821
10/11/2018 13:46:52 : [INFO ] ARMImage name -  :  :  :
10/11/2018 13:46:52 : [INFO ] Base VHD name - disco.vhdx
10/11/2018 13:46:52 : [INFO ] Loading the cycle Data...
10/11/2018 13:46:52 : [INFO ] Loading the test data for VHDX-RESIZE-TO-BIGGER512
10/11/2018 13:46:52 : [INFO ] Loading the test data for VHDX-RESIZE-TO-BIGGER512
10/11/2018 13:55:28 : [INFO ] Checking background cleanup jobs..
10/11/2018 13:55:28 : [INFO ] Removed 0 context files.
10/11/2018 13:55:28 : [INFO ] Cycle Finished.. TC-33821
10/11/2018 13:55:28 : [INFO ]
Test Results Summary
ICA test run on 10/11/2018 13:46:52
VHD under test disco.vhdx
ARM Image under test  :  :  :
Total Executed TestCases         1 (        0 Pass,         1 Fail,         0 Abort)
Total Execution Time(dd:hh:mm) 0:0:8
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VHDX-RESIZE-TO-BIGGER512 : FAIL : 8.59


Logs can be found at .\TestResults\2018-11-10-19-16-51-6878


10/11/2018 13:55:28 : [INFO ] Creating 'HyperV-33821-buildlogs.zip' from '.\TestResults\2018-11-10-19-16-51-6878'
10/11/2018 13:55:29 : [INFO ] D:\Santosh\GitHub\LISAv2\HyperV-33821-buildlogs.zip created successfully.
10/11/2018 13:55:29 : [INFO ] Copied : .\report\report_TC-33821.xml --> .\report\report_TC-33821-junit.xml
10/11/2018 13:55:29 : [INFO ] Analysing results..
10/11/2018 13:55:29 : [INFO ] PASS  : 0
10/11/2018 13:55:29 : [INFO ] FAIL  : 1
10/11/2018 13:55:29 : [INFO ] ABORT : 0
10/11/2018 13:55:29 : [INFO ] LISAv2 exits with code: 1

File: Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+# ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR
+# PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+#
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
```
Comment (santoshx) [inline Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh]: You may need to shorten this License information. See other already merged .sh files.

File: Testscripts/Linux/STOR_VHDXResize_ReadWrite.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0  
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+# ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR
+# PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+#
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
```
Comment (santoshx) [inline Testscripts/Linux/STOR_VHDXResize_ReadWrite.sh]: Same here about License.

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+
+Param([String] $TestParams)
+
+Function Main
+{
+	$tps = Parse-TestParameters -XMLParams $CurrentTestData.TestParameters
+	$vmname = $allVMData.RoleName
+	$hvserver = $allVMData.HyperVHost
+    $ip = $AllVMData.PublicIP
```
Comment (santoshx) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Alignment issue, looks like tab is missing

File: XML/TestCases/FunctionalTests.xml [hunk 4]
```diff
@@ -1144,5 +1144,24 @@
         <Category>Functional</Category>
         <Area>STORAGE</Area>
         <Tags>storage</Tags>
-    </test>
```
Comment (santoshx) [inline XML/TestCases/FunctionalTests.xml]: dont change existing line of code

**Review by santoshx — COMMENTED** [review]
Done

**Review by prashantmhetar — CHANGES_REQUESTED** [review]
Please modify these shell scripts as per the Lisav2 framework

File: Testscripts/Linux/STOR_VHDXResize_ReadWrite.sh [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+#     The test performs the following steps:
+#    1. Creates a file and saves the file size and checksum value
+#    2. Creates a folder on the mounted partition
+#    3. Copies the created file on that specific path
+#    4. Writes, reads and deletes the copied file
+#    5. Deletes the previously created folder 
+#
+########################################################################
+
+ICA_TESTRUNNING="TestRunning"      # The test is running
```
Comment (prashantmhetar) [inline Testscripts/Linux/STOR_VHDXResize_ReadWrite.sh]: Use utils.sh file for TEST status

File: Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+#    1. Make sure we have a constants.sh file.
+#    2. Creates partition
+#    3. Creates filesystem
+#    4. Performs read/write operations
+#    5. Unmounts partition
+#    6. Deletes partition
+#
+########################################################################
+
+ICA_TESTRUNNING="TestRunning"      # The test is running
```
Comment (prashantmhetar) [inline Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh]: Use utils.sh for TEST status

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+	$port = $AllVMData.SSHPort
+
+	try {
+
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Check spaces in PS1 script. Run PSScriptAnalyser to resolve  line alignment

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+		# if host build number lower than 9600, skip test
+		$BuildNumber = Get-HostBuildNumber $hvserver
+
+		if ($BuildNumber -eq 0) {
+			return $false
+		} elseif ($BuildNumber -lt 9600) {
+		    return "Skipped"
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: As we are using try-Block, throw should be used. 
return "Skipped" is not permissable

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+		}
+
+		if ( $tps.ControllerType -eq "IDE" ) {
+			$vmGeneration = Get-VMGeneration $vmname $hvserver
+			if ($vmGeneration -eq 2 ) {
+				LogErr "Generation 2 VM does not support IDE disk, skip test"
+				return "Skipped"
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: same as above

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+				$vhdxDrive = Get-VMHardDiskDrive -VMName $vmname -Controllertype $tps.ControllerType -ControllerNumber `
+								$vhdx.ControllerNumber -ControllerLocation $vhdx.ControllerLocation -ComputerName $hvserver `
+									-ErrorAction SilentlyContinue
+			}
+		}
+
+		if (-not $vhdxDrive) {
+		    LogErr "VM ${vmname} does not have a $tps.ControllerType $vhdxDrive.ControllerNumber lun $vhdxDrive.ControllerLocation drive on ${hvserver}"
+		    $error[0].Exception.Message
+		    return "FAIL"
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Need to use throw

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+		}
+
+		LogMsg "Check if the virtual disk file exists"
+		$vhdPath = $vhdxDrive.Path
+		$vhdxInfo = Get-RemoteFileInfo $vhdPath $hvserver
+		if (-not $vhdxInfo) {
+			LogErr "The vhdx file (${vhdPath} does not exist on server ${hvserver}"
+		    return "FAIL"
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Why so many return "FAIL"?

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+			LogErr "Insufficent disk free space, This test case requires ${tps.NewSize} free, Current free space is $($diskInfo.FreeSpace)"
+		    return "FAIL"
+		}
+
+		$guest_script_path = "./Testscripts/Linux/"
+
+		# Make sure if we can perform Read/Write operations on the guest VM
+		$guest_script = "STOR_VHDXResize_PartitionDisk.sh"
+
+		RemoteCopy -uploadTo $ip -port $port -files $guest_script_path$guest_script -username $user -password $password -upload
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Do we need to remote copy this script? Can it be part of XML file tag

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+		}
+
+		if ($diskSize -ne $newVhdxSize) {
+			LogErr "VM ${vmname} sees a disk size of ${diskSize}, not the expected size of ${newVhdxSize}"
+			return "FAIL"
+		}
+
+		# Make sure if we can perform Read/Write operations on the guest VM
+		if ([int]($newVhdxGrowSize/1gb) -gt 2048) {
+			$guest_script = "STOR_VHDXResize_PartitionDiskOver2TB.sh"
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Is STOR_VHDXResize_PartitionDiskOver2TB.sh already part of Lisav2 framework?

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+		$ret = RunLinuxCmd -ip $ip -port $port -username $user -password $password -command "./$guest_script" -runAsSudo
+		if (-not $ret) {
+			LogErr "Running '${guest_script}'script failed on VM. check VM logs , exiting test case execution"
+			return "FAIL"
+		}
+		LogMsg "The guest sees the new size after resizing ($diskSize)"
+		LogMsg "Info : VHDx Resize - ${TC_COVERED} is Done"
+		return "PASS"
+
+	} catch {
```
Comment (prashantmhetar) [inline Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: If return is used in try { } block do we need catch- finally?

**Review by prashantmhetar — COMMENTED** [review]
I can see test result is Failed. Please perform the testing.

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 (hunk 8 above)
Comment (santoshx) [reply Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Done

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 (hunk 9 above)
Comment (santoshx) [reply Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Done

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 (hunk 13 above)
Comment (santoshx) [reply Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Not needed in these test cases.

File: Testscripts/Windows/STOR_VHDXResize_Basic.ps1 (hunk 14 above)
Comment (santoshx) [reply Testscripts/Windows/STOR_VHDXResize_Basic.ps1]: Done

**Comment by santoshx** [comment]
#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VHDX-RESIZE-TO-BIGGER512 : PASS : 10.23
2. VHDX-RESIZE-TO-BIGGER4096 : PASS : 5.62


Logs can be found at .\TestResults\2018-14-10-23-42-36-3567


10/14/2018 18:28:33 : [INFO ] Creating 'HyperV-39635-buildlogs.zip' from '.\TestResults\2018-14-10-23-42-36-3567'
10/14/2018 18:28:33 : [INFO ] D:\Santosh\GitHub\LISAv2\HyperV-39635-buildlogs.zip created successfully.
10/14/2018 18:28:33 : [INFO ] Copied : .\report\report_TC-39635.xml --> .\report\report_TC-39635-junit.xml
10/14/2018 18:28:33 : [INFO ] Analysing results..
10/14/2018 18:28:33 : [INFO ] PASS  : 2
10/14/2018 18:28:33 : [INFO ] FAIL  : 0
10/14/2018 18:28:33 : [INFO ] ABORT : 0

**Comment by santoshx** [comment]
#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VHDX-RESIZE-TO-BIGGER512 : PASS : 9.59
2. VHDX-RESIZE-TO-BIGGER4096 : PASS : 6.19
3. VHDX-RESIZE-TO-BIGGER512-FIXED-IDE : PASS : 7.58
4. VHDX-RESIZE-TO-BIGGER512-DYNAMIC-IDE : PASS : 7.31
5. VHDX-RESIZE-TO-SMALLER512 : PASS : 6.08
6. VHDX-RESIZE-TO-SMALLER4096 : PASS : 11.92


Logs can be found at .\TestResults\2018-16-10-14-51-35-6425


10/16/2018 10:10:18 : [INFO ] Creating 'HyperV-22331-buildlogs.zip' from '.\TestResults\2018-16-10-14-51-35-6425'
10/16/2018 10:10:19 : [INFO ] D:\Santosh\GitHub\LISAv2\HyperV-22331-buildlogs.zip created successfully.
10/16/2018 10:10:19 : [INFO ] Copied : .\report\report_TC-22331.xml --> .\report\report_TC-22331-junit.xml
10/16/2018 10:10:19 : [INFO ] Analysing results..
10/16/2018 10:10:19 : [INFO ] PASS  : 6
10/16/2018 10:10:19 : [INFO ] FAIL  : 0
10/16/2018 10:10:19 : [INFO ] ABORT : 0
10/16/2018 10:10:19 : [INFO ] LISAv2 exits with code: 0

**Pull request merged**